.gitignore

```diff
@@ -7,3 +7,6 @@ __pycache__
 */*.pyc
 build/
 dist/
+
+# For direnv users.
```

Comment:
Should we gitignore `.idea` and remove those files?

Reply:
That's what I'd do, but maybe @richvdh  had a good reason in https://github.com/matrix-org/python-signedjson/commit/acdb658ad992e6b3bd9a1494d8ec307399bd0830?

Reply:
there's stuff in `.idea` that is meant to be shared. See https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems, and also https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore which is github's suggestion for a `.gitignore` file

Reply:
s/meant to be shared/can be shared/ ;)

Let's tackle what to do with .idea in a separate issue (and thus remove the changes to .idea/* from this PR, unless critical)

Reply:
This is 7ef8b8f.